Unable to build demo_service

Following the instruction here:
https://github.com/includeos/includeos-docker-images#using-the-docker-image-to-build-your-service

I'm trying to build the IncludeOS demo_service:
https://github.com/hioa-cs/IncludeOS/tree/master/examples/demo_service

I've pulled and built the latest docker-images.

I've pulled the latest IncludeOS tree
I changed to the examples/demo_service/ and did ```mkdir build```

NOTE: I needed to also do ```chmod 777 build``` so as docker could write there.

I then ``` cd build``` and run

```docker run --rm -v $(dirname $PWD):/service includeos/includeos-build:v0.11.0-rc.2.01```

But I get compilation errors as below:
```
-- Building for arch x86_64, platform x86_pc
-- Target CPU architecture x86_64
-- Target triple x86_64-pc-linux-elf
-- Using vanilla CPU features: SSE3. CAPABS = -msse3 -mfpmath=sse
-- Looking for drivers in /home/ubuntu/IncludeOS_install/includeos/x86_64/drivers
--      * Found virtionet
-- Looking for plugins in /home/ubuntu/IncludeOS_install/includeos/x86_64/plugins
--      * Found autoconf
-- Configuring done
-- Generating done
-- Build files have been written to: /service/build
[ 40%] Built target config_json
[ 60%] Building CXX object CMakeFiles/service.dir/service.cpp.o
/service/service.cpp:128:58: warning: format specifies type 'unsigned int' but the argument has type 'size_t' (aka 'unsigned long') [-Wformat]
      printf("<Service> @on_write: %u bytes written.\n", written);
                                   ~~                    ^~~~~~~
                                   %zu
In file included from /service/service.cpp:21:
In file included from /home/ubuntu/IncludeOS_install/includeos/api/os:23:
In file included from /home/ubuntu/IncludeOS_install/includeos/api/hw/devices.hpp:22:
In file included from /home/ubuntu/IncludeOS_install/includeos/api/hw/nic.hpp:21:
In file included from /home/ubuntu/IncludeOS_install/includeos/api/hw/../net/buffer_store.hpp:24:
In file included from /home/ubuntu/IncludeOS_install/includeos/api/smp:23:
In file included from /home/ubuntu/IncludeOS_install/includeos/api/delegate:23:
/home/ubuntu/IncludeOS_install/includeos/api/util/delegate.hpp:244:13: error: no matching function for call to object of type '(lambda at /service/service.cpp:105:5)'
                        { return reinterpret_cast<C&>(storage)(std::forward<Args>(args)...); }
                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/ubuntu/IncludeOS_install/includeos/api/util/delegate.hpp:395:3: note: in instantiation of function template specialization 'spec::inplace<32, 16, void, std::__1::shared_ptr<unsigned char>, unsigned long>::inplace<(lambda at /service/service.cpp:105:5), (lambda at /service/service.cpp:105:5)>' requested here
                storage_{ std::forward<T>(val) }
                ^
/service/service.cpp:105:5: note: in instantiation of function template specialization 'delegate<void (std::__1::shared_ptr<unsigned char>, unsigned long), spec::inplace, 32, 16>::delegate<(lambda at /service/service.cpp:105:5), void>' requested here
    [conn] (auto buf)
    ^
/service/service.cpp:105:5: note: candidate function template not viable: requires single argument 'buf', but 2 arguments were provided
1 warning and 1 error generated.
make[2]: *** [CMakeFiles/service.dir/service.cpp.o] Error 1
CMakeFiles/service.dir/build.make:62: recipe for target 'CMakeFiles/service.dir/service.cpp.o' failed
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/service.dir/all' failed
Makefile:127: recipe for target 'all' failed
make[1]: *** [CMakeFiles/service.dir/all] Error 2
make: *** [all] Error 2
``` 

Creating the issue here (builder not in sync with current IncludeOS?) but maybe should have filed against hioa-cs/IncludeOS (maybe example code is not up to date?).

--- a/Dockerfile.build
+++ b/Dockerfile.build
@@ -1,5 +1,5 @@
 # PARENT can be specified when building with --build-arg PARENT=...
-ARG PARENT=includeos/includeos-common:v0.11.0-rc.2.01
+ARG PARENT=includeos/includeos-common:v0.12.0-rc.1
 FROM $PARENT
 
 VOLUME /service

--- a/Dockerfile.grubify
+++ b/Dockerfile.grubify
@@ -1,4 +1,4 @@
-FROM includeos/includeos-common:v0.11.0-rc.2.01 as base
+FROM includeos/includeos-common:v0.12.0-rc.1 as base
 
 FROM ubuntu:xenial
 

--- a/Dockerfile.qemu
+++ b/Dockerfile.qemu
@@ -1,4 +1,4 @@
-FROM includeos/includeos-common:v0.11.0-rc.2.01
+FROM includeos/includeos-common:v0.12.0-rc.1
 
 VOLUME /service/build
 WORKDIR /service/build

--- a/Dockerfile.webserver
+++ b/Dockerfile.webserver
@@ -1,4 +1,4 @@
-FROM includeos/includeos-common:v0.11.0-rc.2.01
+FROM includeos/includeos-common:v0.12.0-rc.1
 
 VOLUME /public
 WORKDIR /home/ubuntu/IncludeOS/examples/acorn

--- a/README.md
+++ b/README.md
@@ -5,11 +5,11 @@ These Docker images let you try out building [IncludeOS](https://github.com/hioa
 ## Building the IncludeOS Docker images
 
 ```
-$ docker build -t includeos/includeos-common:v0.11.0-rc.2.01 -f Dockerfile.common .
-$ docker build -t includeos/includeos-build:v0.11.0-rc.2.01 -f Dockerfile.build .
-$ docker build -t includeos/includeos-qemu:v0.11.0-rc.2.01 -f Dockerfile.qemu .
-$ docker build -t includeos/includeos-grubify:v0.11.0-rc.2.01 -f Dockerfile.grubify .
-$ docker build -t includeos/includeos-webserver:v0.11.0-rc.2.01 -f Dockerfile.webserver .
+$ docker build -t includeos/includeos-common:v0.12.0-rc.1 -f Dockerfile.common .
+$ docker build -t includeos/includeos-build:v0.12.0-rc.1 -f Dockerfile.build .
+$ docker build -t includeos/includeos-qemu:v0.12.0-rc.1 -f Dockerfile.qemu .
+$ docker build -t includeos/includeos-grubify:v0.12.0-rc.1 -f Dockerfile.grubify .
+$ docker build -t includeos/includeos-webserver:v0.12.0-rc.1 -f Dockerfile.webserver .
 ```
 
 ## Using the Docker image to build your service
@@ -17,7 +17,7 @@ $ docker build -t includeos/includeos-webserver:v0.11.0-rc.2.01 -f Dockerfile.we
 ```
 $ cd <my-super-cool-service>
 $ mkdir build && cd build
-$ docker run --rm -v $(dirname $PWD):/service includeos/includeos-build:v0.11.0-rc.2.01
+$ docker run --rm -v $(dirname $PWD):/service includeos/includeos-build:v0.12.0-rc.1
 ```
 
 ## Running a very basic sanity test of your service image
@@ -25,7 +25,7 @@ $ docker run --rm -v $(dirname $PWD):/service includeos/includeos-build:v0.11.0-
 Don't have a hypervisor installed? No problem? Run your service inside QEMU in a Docker container:
 
 ```
-$ docker run --rm -v $PWD:/service/build includeos/includeos-qemu:v0.11.0-rc.2.01 <image_name>
+$ docker run --rm -v $PWD:/service/build includeos/includeos-qemu:v0.12.0-rc.1 <image_name>
 ```
 
 (If the service is not designed to exit on its own, the container must be stopped with `docker stop`.)
@@ -35,7 +35,7 @@ $ docker run --rm -v $PWD:/service/build includeos/includeos-qemu:v0.11.0-rc.2.0
 On macOS, the `boot -g` option to add a GRUB bootloader is not available. Instead, you can use the `includeos-grubify` Docker image. Build your service, followed by:
 
 ```
-$ docker run --rm --privileged -v $(dirname $PWD):/service includeos/includeos-grubify:v0.11.0-rc.2.01 /service/build/<image_name>
+$ docker run --rm --privileged -v $(dirname $PWD):/service includeos/includeos-grubify:v0.12.0-rc.1 /service/build/<image_name>
 ```
 
 ## Building a tiny web server
@@ -43,5 +43,5 @@ $ docker run --rm --privileged -v $(dirname $PWD):/service includeos/includeos-g
 Do you have some web content that you would like to serve, without having to figure out arcane Apache configuration files? Just go to the folder where your web content is located and build a bootable web server:
 
 ```
-docker run --rm -v $PWD:/public includeos/includeos-webserver:v0.11.0-rc.2.01
+docker run --rm -v $PWD:/public includeos/includeos-webserver:v0.12.0-rc.1
 ```

